overlays/nodejs/tags.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Tags represents a set of key-value string pairs to which can be applied
+ * to an AWS resource.
+ */
+export interface Tags {
```

Comment:
It's a little odd to define a type in this library which is never used.  

We could in principle override the type of properties that accept this shape.  Are there a few we should apply this type to as part of the initial PR?

Reply:
Good point, it hadn't occurred to me to go through and replace the usages. I've made a start on doing so, with each resource as a separate commit. If you'd rather they were a single commit for all resources it should be easy enough to squash them.

Reply:
I think all the resources are now done, but there is still some work to do to make the data sources work.